lib/valueflow.cpp

```diff
@@ -3638,7 +3638,9 @@ static std::vector<ValueFlow::LifetimeToken> getLifetimeTokens(const Token* tok,
         while (vartok) {
             if (vartok->str() == "[" || vartok->originalName() == "->" || vartok->isUnaryOp("*"))
                 vartok = vartok->astOperand1();
-            else if (vartok->str() == "." || vartok->str() == "::")
+            else if (vartok->str() == ".")
+                vartok = vartok->astOperand1();
+            else if (vartok->str() == "::")
```

Comment:
I think this change makes sense overall, though there maybe some places expecting a variable(ie `getLifetimeVariable`). So I was always concerned about causing issues, but it looks like only one test failed.

Reply:
Ah wait nevermind, this is changing `a.x` to point to `a` instead of `x`. I thought it was changing it to point to `a.x`. This would be a better change, but I guess there is a lot more test failures.

For a quick fix, it seems like #5970 is a better solution for a quick fix.